lib/deploy/functions/deploy.js

```diff
@@ -1,19 +1,23 @@
 'use strict';
 
+var _ = require('lodash');
 var chalk = require('chalk');
 var RSVP = require('rsvp');
 var tmp = require('tmp');
 var utils = require('../../utils');
 var gcp = require('../../gcp');
 var prepareFunctionsUpload = require('../../prepareFunctionsUpload');
 
+var GCP_REGION = 'us-central1';
+
 tmp.setGracefulCleanup();
 
 module.exports = function(context, options, payload) {
   var _uploadSource = function(source) {
-    return gcp.storage.buckets.acquire(context.projectId, context.functionsBucket)
-    .then(function(bucketName) {
-      return gcp.storage.upload(source.stream, bucketName);
+    return gcp.storage.getUploadUrl(context.projectId, GCP_REGION).then(function(uploadUrl) {
+      _.set(context, 'uploadUrl', uploadUrl);
+      uploadUrl = _.replace(uploadUrl, 'https://storage.googleapis.com', '');
```

Comment:
nit: `_.trimStart` seems more appropriate.

Reply:
Also, do we want to use the `storageOrigin` value instead of the string for production?

Reply:
The alternative is to pass in `origin: ''` while making the api.request since origin will automatically default to admin otherwise.

Reply:
`_.trimStart` removes characters from the string until it encounters a character not in the string, so will not work in this case.